RecoTauTag/HLTProducers/src/HLTPFDiJetCorrCheckerWithDiTau.cc

```diff
@@ -71,48 +71,50 @@ void HLTPFDiJetCorrCheckerWithDiTau::produce(edm::StreamID iSId, edm::Event& iEv
 
   std::set<unsigned int> indices;
 
-  if (pfJets.size() > 1 && !taus.empty()) {
-    for (unsigned int iJet1 = 0; iJet1 < pfJets.size(); iJet1++) {
-      for (unsigned int iJet2 = iJet1 + 1; iJet2 < pfJets.size(); iJet2++) {
-        bool correctComb = false;
-        const reco::PFJet& myPFJet1 = pfJets[iJet1];
-        const reco::PFJet& myPFJet2 = pfJets[iJet2];
-
-        if ((myPFJet1.p4() + myPFJet2.p4()).M() < mjjMin_)
+  for (unsigned int iJet1 = 0; iJet1 < pfJets.size(); iJet1++) {
+    for (unsigned int iJet2 = iJet1 + 1; iJet2 < pfJets.size(); iJet2++) {
+      bool correctComb = false;
+      const reco::PFJet& myPFJet1 = pfJets[iJet1];
+      const reco::PFJet& myPFJet2 = pfJets[iJet2];
+
+      if ((myPFJet1.p4() + myPFJet2.p4()).M() < mjjMin_)
+        continue;
+
+      for (unsigned int iTau1 = 0; iTau1 < taus.size(); iTau1++) {
+        if (reco::deltaR2(taus[iTau1]->p4(), myPFJet1.p4()) < dRmin2_)
+          continue;
+        if (reco::deltaR2(taus[iTau1]->p4(), myPFJet2.p4()) < dRmin2_)
           continue;
 
-        for (unsigned int iTau1 = 0; iTau1 < taus.size(); iTau1++) {
-          if (reco::deltaR2(taus[iTau1]->p4(), myPFJet1.p4()) < dRmin2_)
-            continue;
-          if (reco::deltaR2(taus[iTau1]->p4(), myPFJet2.p4()) < dRmin2_)
+        if (taus.size() == 1) { 
+          if (taus[iTau1]->pt() < extraTauPtCut_) 
```

Comment:
Can we move `taus[iTau1]->pt() < extraTauPtCut_` outside this `if`, and remove the 2nd `taus[iTau1]->pt() < extraTauPtCut_` further below in the `else` ?

Reply:
If we move those statements as you suggest then in the case of two taus the cut is applied to both. It is only intended to be applied to the leading tau, hence its inclusion in the ```if``` as well as the ```else``` statement. In the case of two taus, the pt isn't ordered. If they were pt-ordered, we could put ```taus[iTau1]->pt() < extraTauPtCut_``` outside the if statement and remove the check on the 2nd tau entirely.

Reply:
Okay, thanks for the clarification.